### DISCUSSION TITLE
fabric "Uncaught exception in thread 'main'"

### DISCUSSION THREAD

**Discussion by strikeknight57** [FabricMC/fabric-api#2267] [Player support]
trying to launch fabric or any other version (forge or what ever else) and I get the following error... 

Uncaught exception in thread "main"
net.fabricmc.loader.impl.util.ExceptionUtil$WrappedException: java.io.IOException: error opening C:\Users\logan\AppData\Roaming\.minecraft\.fabric\remappedJars\minecraft-1.18.2-0.14.6\client-intermediary.jar: java.util.zip.ZipException: zip END header not found
	at net.fabricmc.loader.impl.util.ExceptionUtil.wrap(ExceptionUtil.java:51)
	at net.fabricmc.loader.impl.game.patch.GameTransformer.lambda$locateEntrypoints$0(GameTransformer.java:87)
	at net.fabricmc.loader.impl.game.minecraft.patch.EntrypointPatch.process(EntrypointPatch.java:62)
	at net.fabricmc.loader.impl.game.patch.GameTransformer.locateEntrypoints(GameTransformer.java:92)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.initialize(MinecraftGameProvider.java:339)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:142)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.io.IOException: error opening C:\Users\logan\AppData\Roaming\.minecraft\.fabric\remappedJars\minecraft-1.18.2-0.14.6\client-intermediary.jar: java.util.zip.ZipException: zip END header not found
	at net.fabricmc.loader.impl.util.SimpleClassPath.getEntry(SimpleClassPath.java:80)
	at net.fabricmc.loader.impl.game.patch.GameTransformer.lambda$locateEntrypoints$0(GameTransformer.java:78)
	... 6 more
Caused by: java.util.zip.ZipException: zip END header not found
	at java.base/java.util.zip.ZipFile$Source.findEND(ZipFile.java:1469)
	at java.base/java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1477)
	at java.base/java.util.zip.ZipFile$Source.<init>(ZipFile.java:1315)
	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1277)
	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:709)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:243)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:172)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:186)
	at net.fabricmc.loader.impl.util.SimpleClassPath.getEntry(SimpleClassPath.java:78)
	... 7 more

I have no idea what is going on here... can anyone help me out here?

I'm worried it might be something with my system (I'm running on the bleeding  edge of 12th gen intel and ddr5) as I've been getting a lot of BSODs any time that I try and launch the normal version of minecraft. (code clock_Watchdog_timeout) I updated all of the drivers needed for my mother board chipset and what not else and the normal version works now. Now I just get this error above any time I try to launch fabric or forge. I'm at a loss... 

I am running on the newest fabric loader 0.14.6 already (that was a suggestion I saw in a few other posts) and I've made it a point to update all of the mods that I have on my setup.

**Comment by Linguardium** [comment]
Caused by: java.io.IOException: error opening C:\Users\logan\AppData\Roaming.minecraft.fabric\remappedJars\minecraft-1.18.2-0.14.6\client-intermediary.jar: java.util.zip.ZipException: zip

This is telling you that you have a corrupt zip file in your folder. try deleting the .fabric folder from your .minecraft folder and running the game again

**Comment by strikeknight57** [comment]
K, I did that and ended up with a Hypervisor error BSOD... I'm starting to wonder if my OS is corrupted... the only problem there is this is only happening when I try to run anything minecraft related... I haven't had this issue with anything else...

**Reply by Linguardium** [reply]
definitely sounds like hard drive or ram is starting to go. If you know how and your computer has builtin diagnostic testing when you boot to bios, might be worth running them.  A bad hard drive can cause issues with saving/loading jar files, but bad ram can too.  bad ram can cause random BSOD for sure.

**Reply by strikeknight57** [reply]
trouble there is it is all brand new stuff... I guess I will have to go through New Egg's RMA system... that's gonna suck... I actually Build PCs for a living... and at this point I'm pretty sure it's the ram. Less likely it's the ssd... Had to go bleeding edge with my showpiece build lol... guess that's DDR5 for you...

**Comment by jan-Sanku** [comment]
Hi, I'm having a very similar issue. I've tried sfc, MdSched, chkdsk, and updating GPU drivers, but nothing helps. I'm not getting a BSOD though, just an instant reboot (Kernel-Power event ID 41).
Does anyone have any suggestions what kind of test to run or drivers to update? I'm at my wit's end...
It was all literally working fine yesterday, and I've done nothing but watch YT and play MC in the meantime (yes yes I know)

**Reply by Linguardium** [reply]
Check temperatures. Try reducing power load by using on board graphics (remove gpu) or replacing the power supply.

**Reply by jan-Sanku** [reply]
Thanks for the reply. I don't have on board graphics and I also don't have a spare power supply lying around (this is my first desktop PC, replacing an old laptop). My temps are all fine, below 40°C, and don't really have time going up since starting MC crashes the PC instantly. I don't think it's a power issue, as far as I know the "Kernel-Power" event simply means the PC shut down unexpectedly. The same event occurred to me on my old laptop with a regular BSOD.

**Reply by jan-Sanku** [reply]
I'm crashing whenever I try to start a version of Fabric for the first time since this started happening, any version that I've successfully loaded in the past works. Specifically the files generating at or after the log entry with the "optimized datafixer" thing need to be functional. It seems that otherwise fabric tries to build them, which causes the crash.

**Reply by warjort** [reply]
Forget the datafixer, it is a red herring. It is just a log message that happens early in the loading process during initial static classloading.

Your problem is that the fabric loader hasn't finished converting mojang's obfuscated minecraft.jar to fabric's intermediate format because the OS crashed while it tried to do it.
If in the past this did work, the loader will continue to use the already generated version without problem.

Now the crash is not fabric's problem. It should not be possible for java code to crash the OS.
The most likely problems as has already been said are;
* Out-of-date system software - operating system, drivers, bios, etc.
* Faulty hardware

I suggested before you contact your PC supplier or MS, but here is some simple things you could check also:
* Open the case (assuming that does not break your warranty) and check all the cards are sitting correctly and there are no loose connections
* Delete the .fabric folder in your .minecraft so maybe it recreates somewhere else on the physical disk
* Move your .minecraft folder somewhere else on the file system (e.g. try a different OS account/user)
I am assuming that because the crash occurs while fabric is writing to a jar file, it is the filesystem/disk that is problematic.

**Reply by jan-Sanku** [reply]
> the fabric loader hasn't finished [building some jar] because the OS crashed while it tried to do it.
If in the past this did work, the loader will continue to use the already generated version without problem.

I thought as much, though I though perhaps that's the datafixer since it's the last message in the logs, but apparently not.

> Out-of-date system software - operating system, drivers, bios, etc.

My Win10 is always up-to-date. I tried updating my GPU drivers, since they's the only drivers I've ever tried upgrading and I'm not sure what else to try. I honestly have no idea how to upgrade my BIOS since my motherboard is pretty obscure (it's a prebuilt PC, sadly, since the price of my planned build skyrocketed but they had this almost equivalent one for the original price) and I can't even find a manual for it. In case you want to know, it's called b560h6-em by Medion, which is a trusted company here but the motherboard is still slightly obscure. The BIOS itself is an American Megatrends UEFI.

> Faulty hardware

Oh boy I sure hope not...

> Open the case (assuming that does not break your warranty) and check all the cards are sitting correctly and there are no loose connections

If that did break warranty it was worth it, because I'd never get a PC and never even open it up. The built-in storage isn't good enough, so I had to install it myself. Before you ask, .minecraft and Java and everything else related to this are all installed on the original built-in SSD. (I did downgrade from Win11 to Win10 for personal reasons though)
I checked the RAM sticks already since I also had some weird memory leak error that was seemingly caused by my shaders, but doesn't occur anymore, but I can check other connections tomorrow. I imagine especially the M.2 slots would be of interest? I can check the CPU and GPU too but that would require a lot of screwing around, literally.
(that memory thing was weird though, for some reason my pagefile was limited to 10 GB so I set it back to auto, and now when I play on 32 chunks with shaders on my game literally uses over 30 GB of memory total, 16 GB of that being my 2x8 dedicated RAM and the other half being my in pagefile... what?)

> Delete the .fabric folder in your .minecraft so maybe it recreates somewhere else on the physical disk

Tried that, no avail.

> Move your .minecraft folder somewhere else on the file system (e.g. try a different OS account/user)
I am assuming that because the crash occurs while fabric is writing to a jar file, it is the filesystem/disk that is problematic.

I will try that and report back. This reminds me that I could also try it inside a VM just because why tf not, I'm desperate at this point and any data I can gather might be helpful.

**Reply by jan-Sanku** [reply]
Using a different user account still crashes. Using a VM doesn't, not surprising but worth a shot I guess. Like I said, I'll check hardware to the best of my knowledge tomorrow, any help on the BIOS thing would be appreciated in the meantime.

**Reply by warjort** [reply]
> any help on the BIOS thing would be appreciated in the meantime.

Typically:
You will need to search for whoever manufactures your bios.
They will let you download a file that you can copy/unzip onto a usb stick.
Then during boot, you press a key to enter bios setup and install the new bios.

Make sure the bios is actually an update/later version.
And make sure you have a backup of the old bios in case the update fails or causes problems.

Again asking your PC supplier is probably the best way to go.

**Reply by warjort** [reply]
> Using a different user account still crashes. Using a VM doesn't

That's weird because presumably both are just using different parts of the same disk?

Maybe your downgrade to Windows 10 left some rubbish lying around (registry setting, wrong driver, etc.) that is the cause the problem?
Something that is not present for a fresh OS install in a VM.

**Reply by jan-Sanku** [reply]
> Again asking your PC supplier is probably the best way to go.

I'll try that.

> That's weird because presumably both are just using different parts of the same disk?

Yes, and everything else disk related is working just fine.

> Maybe your downgrade to Windows 10 left some rubbish lying around (registry setting, wrong driver, etc.) that is the cause the problem?

No I wiped the entire thing because I wanted to get rid of the preinstalled bloatware. Reformatted the C: drive and then did a fully clean Win10 install. It's been working flawlessly for months now, installing and running big games without issue. The only weird thing that happened was MC crashing with shaders active, which at 1 point also prompted a windows error about some DLL being broken which was detected and successfully fixed by `sfc /scannow` so that seems unrelated. That was before I upgraded to 1.19, meaning that Fabric has been able to build its jars after that short incident.

**Reply by jan-Sanku** [reply]
Now I don't know if there were any underlying issues that will cause more trouble down the road, but it seems that Riot Vanguard is the reason for the crashes. How? Not the slightest clue, but deactivating it mysteriously allowed me to run Fabric without issue, and doing a clean reinstall of Valorant seems to be a permanent fix so far. It kinda makes sense since Vanguard has kernel level access, but I don't know why it crashed when Fabric tried to install itself. Regardless, thank you all for your help!
